### DISCUSSION TITLE
pwntools cannot handle SIGSEGV in my Archlinux

### DISCUSSION THREAD

**Discussion by Nambers** [Gallopsled/pwntools#2340] [Q&A] [answered]
Hi I don't know why pwntools(`pwntools 4.11.1`) in my Arch cannot handle SIGSEGV (on both `process` and `gdb.debug`)
![image](https://github.com/Gallopsled/pwntools/assets/35139537/e1fbdf42-1fc1-4e28-ab2e-56754ba8d15f)
And my script:
![image](https://github.com/Gallopsled/pwntools/assets/35139537/ac892a8e-c474-4db3-ad65-722b0b0fcb90)

**Comment by peace-maker** [comment, marked as answer]
I assume your exploit terminates and takes the debugger with it. Try adding `conn.interactive()` at the end, to keep the python process around.

**Reply by Nambers** [reply]
Yes you are right! Have a while not using pwntools and I totally forgot about that...
btw, is there a way to use pwntool's gdb inside of vscode?
![image](https://github.com/Gallopsled/pwntools/assets/35139537/366e1ec2-b1ba-407b-ba79-903d5accdda1)

**Reply by peace-maker** [reply]
There's an open pull request https://github.com/Gallopsled/pwntools/pull/2329 which at least suppresses that error. Some other terminal rule might work then.

**Reply by Nambers** [reply]
Yes that fix works on my Archlinux

**Comment by RocketMaDev** [comment]
Consider about ipython. Ipython wouldn't kill the process when exploit terminates. And sometimes I want to check the remaining output of `conn` but using python to run the exploit requires you to add a line at the bottom, but with Ipython, just type `conn.recv()`. Use `%load` to load your payload.

By the way, I'm also a Arch Linux user, and tmux is great when you want to debug. It may be an alternative that open tmux in vscode so pwntools treat shell as tmux.
